Fw/Types/Serializable.hpp

```diff
@@ -40,154 +45,1218 @@ class Serializable {
+    //! data in the format produced by serializeTo().
+    //!
+    //! \param buffer Reference to the SerialBufferBase from which data will be deserialized
+    //! \param mode Endianness mode for deserialization (default is Endianness::BIG)
+    //! \return SerializeStatus indicating the result of the operation
+    virtual SerializeStatus deserializeFrom(SerialBufferBase& buffer, Endianness mode = Endianness::BIG) = 0;
 
     // ----------------------------------------------------------------------
     // Legacy methods for backward compatibility
     // ----------------------------------------------------------------------
 
-    DEPRECATED(SerializeStatus serialize(SerializeBufferBase& buffer) const,
-               "Use serializeTo(SerializeBufferBase& buffer) instead") {
+    DEPRECATED(SerializeStatus serialize(SerialBufferBase& buffer) const,
+               "Use serializeTo(SerialBufferBase& buffer) instead") {
         return this->serializeTo(buffer);
     }
 
-    DEPRECATED(SerializeStatus deserialize(SerializeBufferBase& buffer),
-               "Use deserializeFrom(SerializeBufferBase& buffer) instead") {
+    DEPRECATED(SerializeStatus deserialize(SerialBufferBase& buffer),
+               "Use deserializeFrom(SerialBufferBase& buffer) instead") {
         return this->deserializeFrom(buffer);
     }
 
 #if FW_SERIALIZABLE_TO_STRING || FW_ENABLE_TEXT_LOGGING || BUILD_UT
-    virtual void toString(StringBase& text) const;  //!< generate text from serializable
+    //! \brief Generate a human-readable string representation of this object
+    //!
+    //! This method converts the object's data into a textual representation
+    //! that can be used for logging, debugging, or display purposes. The
+    //! exact format of the string may vary depending on the implementation
+    //! of the derived class.
+    //!
+    //! \param text Reference to a StringBase object where the text representation will be stored
+    virtual void toString(StringBase& text) const;
 #endif
 
 #ifdef BUILD_UT
     friend std::ostream& operator<<(std::ostream& os, const Serializable& val);
 #endif
 
   protected:
-    Serializable();           //!< Default constructor
-    virtual ~Serializable();  //!< destructor
+    //! \brief Default constructor
+    //!
+    //! Initializes a Serializable object with default values.
+    Serializable();
+
+    //! \brief Virtual destructor
+    //!
+    //! Ensures proper cleanup of derived classes.
+    virtual ~Serializable();
 };
 
-class SerializeBufferBase {
+class SerialBufferBase {
+  public:
+    //! \brief Virtual destructor
+    //!
+    //! Ensures proper cleanup of derived classes.
+    virtual ~SerialBufferBase();
+
+    // Serialization for built-in types
+
+    //! \brief Serialize an 8-bit unsigned integer value
+    //!
+    //! This method serializes a single 8-bit unsigned integer value into the buffer.
+    //! The endianness of the serialization can be controlled via the mode parameter.
+    //!
```

Comment:
Maybe note that this is done for compatibility (endianness) as U8s don't have endianness.